libs/s25main/ingameWindows/IngameWindows.h

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2005 - 2020 Settlers Freaks (sf-team at siedler25.org)
+//
+// This file is part of Return To The Roots.
+//
+// Return To The Roots is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// Return To The Roots is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Return To The Roots. If not, see <http://www.gnu.org/licenses/>.
+
+#pragma once
+
+#include "iwAIDebug.h"
+#include "iwAction.h"
+#include "iwBaseWarehouse.h"
+#include "iwBuildOrder.h"
+#include "iwBuilding.h"
+#include "iwBuildingProductivities.h"
+#include "iwBuildingSite.h"
+#include "iwBuildings.h"
+#include "iwChat.h"
+#include "iwDiplomacy.h"
+#include "iwDistribution.h"
+#include "iwEconomicProgress.h"
+#include "iwEndgame.h"
+#include "iwHQ.h"
+#include "iwHarborBuilding.h"
+#include "iwInventory.h"
+#include "iwMainMenu.h"
+#include "iwMapDebug.h"
+#include "iwMerchandiseStatistics.h"
+#include "iwMilitary.h"
+#include "iwMilitaryBuilding.h"
+#include "iwMinimap.h"
+#include "iwMusicPlayer.h"
+#include "iwOptionsWindow.h"
+#include "iwPostWindow.h"
+#include "iwRoadWindow.h"
+#include "iwSave.h"
+#include "iwShip.h"
+#include "iwSkipGFs.h"
+#include "iwStatistics.h"
+#include "iwTextfile.h"
+#include "iwTools.h"
+#include "iwTrade.h"
+#include "iwTransport.h"
+#include "iwVictory.h"
+#include "gameData/const_gui_ids.h"
+#include <map>
+#include <string>
+
+struct PersistentWindowData
+{
+    std::string name;
+    std::string optionName;
```

Comment:
I want to also save whether the minimap is extended or not. Since it's only used there, maybe I should just make a separate option for minimap extendedness in the ingameOptions category of the settings file.

Reply:
I changed this as in my suggestion in the previous comment.